File: crates/ih-muse-python/src/proto/metric.rs
// crates/ih-muse-python/src/proto/metric.rs

use pyo3::prelude::*;

use super::PyMetricDefinition;
use ih_muse_proto::MetricDefinition as RustMetricDefinition;

#[pymethods]
impl PyMetricDefinition {
    #[new]
    pub fn __init__(code: String, name: String, description: String) -> PyResult<Self> {
        let ekr = RustMetricDefinition::new(code, name, description);
        Ok(Self::from(ekr))
    }
}

File: py-ih-muse/ih_muse/proto/metric.py
# py-ih-muse/ih_muse/muse/muse.py

from ih_muse.ih_muse import PyMetricDefinition


class MetricDefinition:
    _metric_def: PyMetricDefinition

    def __init__(self, code: str, name: str, description: str) -> None:
        self._metric_def = PyMetricDefinition(code, name, description)
